src/Parser/IntlMoneyParser.php

```diff
@@ -45,23 +45,30 @@ public function parse($money, $forceCurrency = null)
             throw new ParserException('Formatted raw money should be string, e.g. $1.00');
         }
 
-        $currencyCode = null;
-        $decimal = $this->formatter->parseCurrency($money, $currencyCode);
-
-        if (null !== $forceCurrency) {
-            $currencyCode = $forceCurrency;
-        }
-
-        $currency = new Currency($currencyCode);
+        $currency = null;
+        $decimal = $this->formatter->parseCurrency($money, $currency);
 
         if (false === $decimal) {
             throw new ParserException(
                 'Cannot parse '.$money.' to Money. '.$this->formatter->getErrorMessage()
             );
         }
 
-        $decimal = (string) $decimal;
+        if (null !== $forceCurrency) {
+            $currency = $forceCurrency;
+        }
+
+        /*
+         * This conversion is only required whilst currency can be either a string or a
+         * Currency object.
+         */
+        if (!$currency instanceof Currency) {
+            $currency = new Currency($currency);
+        }
+
         $subunit = $this->currencies->subunitFor($currency);
+
+        $decimal = (string) $decimal;
```

Comment:
This doesn't seem to be related to the PR. Can you please explain?

Reply:
Nothing changed here - I swapped the order of these two following lines, as it made more sense to me that $decimal would be used immediately after being cast;
`$decimal = (string) $decimal;` and
`$subunit = $this->currencies->subunitFor($currency);`

There's no real benefit to doing this though, so I'll flip it back around so as not to add confusion in the git history.